cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -97,6 +97,26 @@ int32_t gdv_fn_populate_varlen_vector(int64_t context_ptr, int8_t* data_ptr,
   offsets[slot + 1] = offset + entry_len;
   return 0;
 }
+
+void gdv_fn_dec_from_string(const char* in, int32_t in_length,
+                            int32_t* precision_from_str, int32_t* scale_from_str,
+                            int64_t* dec_high_from_str, uint64_t* dec_low_from_str) {
+  arrow::Decimal128 dec;
+  auto status = arrow::Decimal128::FromString(std::string(in, in_length), &dec,
+                                              precision_from_str, scale_from_str);
+  *dec_high_from_str = dec.high_bits();
```

Comment:
we seem to be ignoring status..if it is not ok then we should set error in context and return.
else this is likely to crash..